mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1179,10 +1180,8 @@ func expandReplicationSpecs(d *schema.ResourceData) ([]matlas.ReplicationSpec, e
 				original, _ := d.GetChange("replication_specs")
 				for _, s := range original.(*schema.Set).List() {
 					oldSpecs := s.(map[string]interface{})
-					if spec["zone_name"].(string) == cast.ToString(oldSpecs["zone_name"]) {
-						id = oldSpecs["id"].(string)
-						break
-					}
+					id = oldSpecs["id"].(string)
```

Comment:
if  `replication_specs` has only one element, shouldn't it have the max len set to `1` too?

Reply:
this resource can have multiple replication specs per cluster per zone